urllib3/util/retry.py

```diff
@@ -361,6 +371,8 @@ def increment(self, method=None, url=None, response=None, error=None,
             # status_forcelist and a the given method is in the whitelist
             cause = ResponseError.GENERIC_ERROR
             if response and response.status:
+                if status_count is not None:
+                    status_count -= 1
```

Comment:
Actually, I moved this in here from outer block. It would have incremented counter also for `ProxyError` and `NewConnectionError` (not sure if it can really get thrown) at least.